src/cloudformation_cli_python_lib/scheduler.py

```diff
@@ -0,0 +1,58 @@
+import json
+import logging
+from datetime import datetime, timedelta
+from uuid import uuid4
+
+# boto3 doesn't have stub files
+from boto3 import Session  # type: ignore
+
+from .utils import HandlerRequest, KitchenSinkEncoder
+
+LOG = logging.getLogger(__name__)
+
+
+class CloudWatchScheduler:
+    def __init__(self, boto3_session: Session):
+        self.client = boto3_session.client("events")
+
+    def reschedule_after_minutes(
+        self, function_arn: str, minutes_from_now: int, handler_request: HandlerRequest
+    ) -> None:
+        cron = self._min_to_cron(max(minutes_from_now, 1))
+        uuid = uuid4().hex
+        rule_name = f"reinvoke-handler-{uuid}"
+        target_id = f"reinvoke-target-{uuid}"
+        handler_request.requestContext["cloudWatchEventsRuleName"] = rule_name
+        handler_request.requestContext["cloudWatchEventsTargetId"] = target_id
+        json_request = json.dumps(handler_request.serialize(), cls=KitchenSinkEncoder)
+        LOG.info("Scheduling re-invoke at %s (%s)", cron, uuid)
+        self.client.put_rule(Name=rule_name, ScheduleExpression=cron, State="ENABLED")
+        self.client.put_targets(
+            Rule=rule_name,
+            Targets=[{"Id": target_id, "Arn": function_arn, "Input": json_request}],
+        )
+
+    def cleanup_cloudwatch_events(self, rule_name: str, target_id: str) -> None:
+        try:
+            if target_id and rule_name:
+                self.client.remove_targets(Rule=rule_name, Ids=[target_id])
+        except Exception as e:  # pylint: disable=broad-except
```

Comment:
I don't know if it would be better in this case, but you could catch botocore.exceptions.ClientError here.